test/service/test_response.py

```diff
@@ -2613,76 +2625,117 @@ def tearDownClass(cls):
+                self.CellCounts(None, {'brain': 10_000}),
+            ],
+            'effectiveCellCount': [
+                self.CellCounts(3589, {'brain': 1}),
+                self.CellCounts(None, {'Brain': 6210}),
+                self.CellCounts(None, {'presumptive gut': 3869, 'endoderm': 3869}),
+                self.CellCounts(None, {'brain': 10_000}),
+                self.CellCounts(88_000, {'mouth mucosa': None}),
+            ]
+        }
+        for facet in test_data:
+            for ascending in (True, False):
+                with self.subTest(facet=facet, ascending=ascending):
+                    params = {
+                        'catalog': self.catalog,
+                        'sort': facet,
+                        'order': 'asc' if ascending else 'desc'
+                    }
+                    url = self.base_url.set(path='/index/projects', args=params)
+                    response = requests.get(str(url))
+                    response.raise_for_status()
+                    response = response.json()
+                    actual = self.CellCounts.from_response(response)
+                    if ascending:
+                        expected = test_data[facet]
+                    else:
+                        expected = list(reversed(test_data[facet]))
+                    self.assertEqual(actual, expected)
 
-    def test_sorting_by_cell_count(self):
+    def test_filter_by_cell_count(self):
         """
-        Verify sorting by 'cellCount' sorts the documents based on the total
-        number of cells in each document, using the sum of total cells when a
-        document contains more than one cell suspension inner entity.
+        Verify filtering projects by the total cell suspension cell count, the
+        project cell count, and the effective cell count.
         """
-        ascending_results = [
-            (1, 1),
-            (1, 349),
-            (1, 6210),
-            (2, 7738),
-            (1, 10000)
-        ]
-        for ascending in (True, False):
-            with self.subTest(ascending=ascending):
+        test_data = {
+            'cellCount': {
+                'within': [3000, 8000],
+                'results': [
+                    self.CellCounts(None, {'Brain': 6210}),
+                    self.CellCounts(None, {'presumptive gut': 3869, 'endoderm': 3869}),
+                ],
+            },
+            'projectEstimatedCellCount': {
+                'within': [6000, 100_000],
+                'results': [
+                    self.CellCounts(88_000, {'mouth mucosa': None}),
+                ],
+            },
+            'effectiveCellCount': {
+                'within': [3000, 7000],
+                'results': [
+                    self.CellCounts(3589, {'brain': 1}),
+                    self.CellCounts(None, {'Brain': 6210}),
+                ],
+            },
+        }
+        for facet, filter_data in test_data.items():
```

Comment:
Everything is data, we have better, more descriptive names for the elements of a filter.

Reply:
Facets are fields that occur in the termFacets part of the response.